app/controllers/users/omniauth_callbacks_controller.rb

```diff
@@ -1,6 +1,9 @@
 # frozen_string_literal: true
 
 class Users::OmniauthCallbacksController < Devise::OmniauthCallbacksController
+  before_action :gitlab, only: %i[show], constraints: lambda {
```

Comment:
why do we need before action here ?

Reply:
Why do we need to check that for every action action, we need to check that for gitlab action only

Reply:
I have added the commit!
Kindly review.